code/game/machinery/ecmo.dm

```diff
@@ -0,0 +1,594 @@
+			if(blood_message_sent)
+				add_overlay("light_yellow[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("iv_out[tipped ? "_tipped" : ""]")
+	if(tank)
+		add_overlay("tank_oxy[tipped ? "_tipped" : ""]")
+
+		var/tank_level = 2
+		switch(tank.percent())
+			if(-INFINITY to 4)	tank_level = 0
+			if(05 to 19)		tank_level = 1
+			if(20 to 39)		tank_level = 2
+			if(40 to 59)		tank_level = 3
+			if(60 to 79)		tank_level = 4
+			if(80 to 90)		tank_level = 5
+			if(91 to INFINITY)	tank_level = 6
+		add_overlay("[tank.gauge_icon][tank_level][tipped ? "_tipped" : ""]")
+	if(breath_mask)
+		if(breather)
+			add_overlay("arterial_line_on[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("arterial_line_off[tipped ? "_tipped" : ""]")
+	if(panel_open)
+		add_overlay("panel_open[tipped ? "_tipped" : ""]")
+
+/obj/machinery/ecmo/process()
+	breather_process()
+	attached_process()
+
+/obj/machinery/ecmo/proc/breather_process()
+	if(breather)
+		if(!breather.Adjacent(src))
+			src.visible_message(SPAN_WARNING("\The artery line snaps back into \the [src]!"))
+			breath_mask_rip()
+			return
+		if(!tank)
+			return
+		if(breather.species.flags & NO_BREATHE)
+			return
+
+		if(!vein)
+			src.visible_message(SPAN_WARNING("The loop is open (no vein line), disconnecting the artery line!"))
+			breath_mask_rip(TRUE)
+			return
+
+		if(valve_open && attached)
+			var/obj/item/organ/internal/brain/B = breather.internal_organs_by_name[BP_BRAIN]
+			var/obj/item/organ/internal/lungs/L = breather.internal_organs_by_name[BP_LUNGS]
+			var/obj/item/organ/internal/heart/H = breather.internal_organs_by_name[BP_HEART]
+
+			if(B.damage >= brain_damage_pinner && breather.get_brain_result())
+				B.damage = brain_damage_pinner
+
+			if(L.get_oxygen_deprivation() >= 70)
+				L.remove_oxygen_deprivation(10)
+
+			if(H.pulse <= PULSE_SLOW)
+				H.pulse = PULSE_SLOW
+
+			if(tank.air_contents.return_pressure() <= 300)
+				src.visible_message(SPAN_WARNING("\The [src] buzzes, automatically closing the valve due to the lack of tank pressure."))
+				playsound(src, 'sound/machines/buzz-two.ogg', 50)
+				valve_open = FALSE
+				return
+
+			tank.remove_air(0.05) //Not sure what amount those things have, but the ECMO should consume a significant amount of air and must be balanced this way for the sense of urgency
+
+			update_icon()
+
+/obj/machinery/ecmo/proc/attached_process()
+	if(attached)
+		if(!attached.Adjacent(src))
+			iv_rip()
+			update_icon()
+			return
+		if(!beaker)
+			return
+		if(!attached.dna)
+			return
+		if(NOCLONE in attached.mutations)
+			return
+		if(attached.species.flags & NO_BLOOD)
+			return
+
+		if((beaker.reagents.has_reagent(/decl/reagent/blood) || beaker.reagents.has_reagent(/decl/reagent/saline)) && attached.get_blood_volume() >= 100) // Do not add blood if the blood volume is >= 100%
+			update_icon()
+		else
+			if(beaker.reagents.total_volume > 0)
+				beaker.reagents.trans_to_mob(attached, transfer_amount, CHEM_BLOOD)
```

Comment:
Obsoletes IV stands. ECMO and IV stand should be seperate machines for medical use; the ECMO shouldn't just be an IV Stand+

Reply:
Zeng-hu obsoletes the scanners that you normally have in hand, doesn't mean you don't have those anymore, as this machine is much much more expensive than a simple IV stand, and more difficult to use too, I'm not sure what the suggestion is?